### ISSUE TITLE
Problem with loss graphs

### ISSUE THREAD

**Issue by Sangerspyro** [ildoonet/tf-pose-estimation#577]
Hello @ildoonet ,
Thank you very much for your wonderful code!
I have a question regarding training CMU model. I trained the model for 2.76 epochs/21000 steps. When I open the checkpoint files using Tensorboard, it just shows two steps in the loss graph.

```{r, attr.output='style="max-height: 100px;"'}
TensorBoard 1.14.0 at http://linuxuser-System-Product-Name:6006/ (Press CTRL+C to quit)
W0301 15:59:10.038482 140565166167808 plugin_event_accumulator.py:294] Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events.  Overwriting the graph with the newest event.
W0301 15:59:10.440781 140565166167808 plugin_event_accumulator.py:302] Found more than one metagraph event per run. Overwriting the metagraph with the newest event.
W0301 15:59:10.594236 140565166167808 plugin_event_accumulator.py:294] Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events.  Overwriting the graph with the newest event.
W0301 15:59:10.601729 140565166167808 plugin_event_accumulator.py:302] Found more than one metagraph event per run. Overwriting the metagraph with the newest event.
I0301 15:59:16.557330 140565085681408 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:16] "GET / HTTP/1.1" 200 -
I0301 15:59:17.619912 140565085681408 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:17] "GET /font-roboto/oMMgfZMQthOryQo9n22dcuvvDin1pK8aKteLpeZ5c0A.woff2 HTTP/1.1" 200 -
I0301 15:59:18.105598 140565068896000 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/environment HTTP/1.1" 200 -
I0301 15:59:18.108494 140565085681408 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/plugins_listing HTTP/1.1" 200 -
I0301 15:59:18.110013 140564783675136 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/runs HTTP/1.1" 200 -
I0301 15:59:18.112126 140564792067840 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/experiments HTTP/1.1" 200 -
I0301 15:59:18.117464 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /font-roboto/RxZJdnzeo3R5zSexge8UUZBw1xU1rKptJj_0jans920.woff2 HTTP/1.1" 200 -
I0301 15:59:18.237392 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /font-roboto/d-6IYplOFocCacKzxwXSOJBw1xU1rKptJj_0jans920.woff2 HTTP/1.1" 200 -
I0301 15:59:18.372143 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/plugin/scalars/tags HTTP/1.1" 200 -
I0301 15:59:18.405688 140564783675136 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/runs HTTP/1.1" 200 -
I0301 15:59:18.406138 140565068896000 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/environment HTTP/1.1" 200 -
I0301 15:59:18.407073 140564792067840 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/experiments HTTP/1.1" 200 -
I0301 15:59:18.409022 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/plugins_listing HTTP/1.1" 200 -
I0301 15:59:18.694979 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/plugin/scalars/scalars?tag=loss&run=.&experiment= HTTP/1.1" 200 -
I0301 15:59:18.697647 140564766889728 _internal.py:113] ::ffff:127.0.0.1 - - [01/Mar/2020 15:59:18] "GET /data/plugin/scalars/scalars?tag=loss_lastlayer&run=.&experiment= HTTP/1.1" 200 -
```
Tensorboard Graph: ![Alt Text](https://i.imgur.com/52YNRKl.png)

Tensorboard Images: ![Alt Text](https://i.imgur.com/76Idxu7.png)

Training screenshot: ![Alt Text](https://i.imgur.com/BC9rTog.png)

I trained it on the following computer:
* NVIDIA RTX 2060 Super
* 16GB RAM
* UBUNTU 18.04 LTS
* Tensorflow 1.14
* CUDA 10.0
* cuDNN 7.4.2

The command I used to train
`python tf_pose/train.py --model=cmu --batchsize=16 --lr=0.00004 --gpus=1`
I changed the input image size to 272x272 (since 432x368 wasn't running on my PC)

No matter what I change (learning rate, batch size, and input image resolution), it is always giving me this output on Tensorboard.
Any kind of help would be appreciated!

**Comment by mgarbade** [comment]
The `summary_writer` of tensorboard expects the `global_step` (gs) as input argument not the epoch, this is where this issue is coming from

**Comment by Sangerspyro** [comment]
I was able to make it write every 500 steps instead of every epoch by doing the following to train.py:

<ins>Line 207:</ins>
I changed from
```{r, attr.output='style="max-height: 100px;"'}
last_log_epoch1 = last_log_epoch2 = -1
```
to:
```{r, attr.output='style="max-height: 100px;"'}
last_log_epoch1 = last_log_epoch2 = -1
last_log_global_step1 = last_log_global_step2 = -1
```



<ins>Line 223:</ins>
I changed from
```{r, attr.output='style="max-height: 100px;"'}
if last_log_epoch1 < curr_epoch:
                    file_writer.add_summary(summary, curr_epoch)
                    last_log_epoch1 = curr_epoch
```
to:
```{r, attr.output='style="max-height: 100px;"'}
if last_log_global_step1 < gs_num:
                    file_writer.add_summary(summary, gs_num)
                    last_log_global_step1 = gs_num
```



<ins>Line 286:</ins>
I changed from
```{r, attr.output='style="max-height: 100px;"'}
if last_log_epoch2 < curr_epoch:
                    file_writer.add_summary(summary, curr_epoch)
                    last_log_epoch2 = curr_epoch
```
to:
```{r, attr.output='style="max-height: 100px;"'}
if last_log_global_step2 < gs_num:
                    file_writer.add_summary(summary, gs_num)
                    last_log_global_step2 = gs_num
```
This worked for me (I am not sure if this is the right way to do it though)
